### PR TITLE
Adds vectorized time integration to pyoptmat

### PR DESCRIPTION
**Pull request by reverendbedford**
Vectorizes time integration by chunking multiple time steps.

### CODE REVIEW

**Comment by tianjuchen** [comment]
I think we don't need chunk-time-sized temperature dependent scaling of the model parameters. I removed the chunk size in piecewise scaling function, which is working fine on my toy example.

File: examples/structural-inference/tension-temperature/infer.py
Comment (tianjuchen) [inline examples/structural-inference/tension-temperature/infer.py:1]: wrong parameters names

File: examples/structural-inference/tension-temperature/infer.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        )
+        model = models.InelasticModel(
+                E,
+                flowrule
+        )
+
+        return models.ModelIntegrator(model, **kwargs).to(device)
+    
+    # Setup initial guesses
+    names = ["n", "eta", "theta", "tau"]
```
Comment (tianjuchen) [inline examples/structural-inference/tension-temperature/infer.py:71]: names = ["n_vals", "eta_vals", "theta_vals", "tau_vals"]

File: pyoptmat/temperature.py
Comment (tianjuchen) [inline pyoptmat/temperature.py:1]: I guess all the temperature scaling function need to be adjusted if temperature varies as time.

**Review by tianjuchen — CHANGES_REQUESTED** [review]

File: examples/structural-inference/tension-temperature/infer.py
Comment (reverendbedford) [reply examples/structural-inference/tension-temperature/infer.py:1]: These don't have to match the actual parameter names -- they are just for output.

File: pyoptmat/temperature.py
Comment (reverendbedford) [reply pyoptmat/temperature.py:1]: They all should work for the temperature-dependent case now.

**Pull request merged**